model/dataset-filter/filterOverview/filter-overview.go

```diff
@@ -12,20 +12,23 @@ type Page struct {
 // FilterOverview ...
 type FilterOverview struct {
 	Dimensions                 []Dimension   `json:"dimensions"`
+	UnsetDimensions            []string      `json:"unset_dimensions"`
 	PreviewAndDownload         Link          `json:"preview_and_download"`
 	PreviewAndDownloadDisabled bool          `json:"preview_and_download_disabled"`
 	ClearAll                   Link          `json:"clear_all"`
 	Cancel                     Link          `json:"cancel"`
 	IsLatestVersion            bool          `json:"is_latest_version"`
 	LatestVersion              LatestVersion `json:"latest_version"`
 	DatasetTitle               string        `json:"dataset_title"`
+	HasError                   bool          `json:"has_error"`
```

Comment:
Not a 'change something' comment. 

Just wondering how we are using this, if there is an error I imagine this is true, do we show a generic error on the page 'something went wrong' or do we provide more details?

Wondering if we do provide more details and there could be more errors triggered at the same time we could have an array of errors instead. But if we are just going to have a generic one or something like that at all times in the future then this makes sense.

Reply:
Cheers,

So my two cents if I understand this correctly, if this is going to be the only error we will show on this page then yer I think that alternative name might be more clear (saves people misusing it in the future or having to dig around to see what it is).

If there are more potential errors that could happen on the page then I would suggest an error struct of sorts is made then referenced in the filterOverviewStruct, that error struct can just be a bunch of booleans such as `HasUnsetDimensions` (right now it might just be the one) this way it is open to expansion in the future rather than the main FilterOverview struct being a monolithic struct with various bools for errors mixed in with other values.

Or maybe we are naming it (HasError) based on how it is used rather than what it represents, if it is to store a boolean on whether or not there are UnsetDimensions maybe we do just call it something like NoDimensionsSet

Reply:
I've gone with `HasUnsetDimensions`. 

Agree that multiple errors should be handled with an error struct. I've not come across anymore while developing this solution. The only thing you can do on the page is add some options to filter by (clicking a link) or apply the filters (submit). The work is to get around that if you disabled JS you can skip adding the filters as there was no mechanism to check that filters had been applied. So you could argue that there should be a generic error handler that can take multiple errors but at this stage I think it's over engineering the problem. If more problems are found and there needs to be more robust form validation on this page we can definitely revisit and add a better error handling solution.

Reply:
Have approved. I wouldn't necessarily agree that it is over-engineering as we have been bitten in the past by models that have become cumbersome which has lead to duplications being created that then caused bugs and later had to be refactored which took some time.

Reply:
Thanks. I think that part of the issue is that when reviewing the models repo you are detached from the implementation. Hopefully the work to have the specific models needed in the calling service we are planning on bringing into sprint will eventually mitigate this issue. I am happy to show you how I have implemented it and why I believe that it would cause over-engineering as I would have to do significant refactoring to effectively use an error struct. I agree that by adding bools you can have a situation where the model becomes bloated. For example, I believe that `PreviewAndDownloadDisabled` is probably a good candidate for removal as at no point is the preview and download button disabled.